Util/src/main/java/io/deephaven/util/thread/ThreadInitializationFactory.java

```diff
@@ -0,0 +1,44 @@
+package io.deephaven.util.thread;
+
+import io.deephaven.configuration.Configuration;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Extension point to allow threads that will run user code from within the platform to be controlled by configuration.
+ */
+public interface ThreadInitializationFactory {
+    /* private */ String[] CONFIGURED_INITIALIZATION_TYPES =
+            Configuration.getInstance().getStringArrayFromProperty("thread.initialization");
+    /* private */ List<ThreadInitializationFactory> INITIALIZERS = Arrays.stream(CONFIGURED_INITIALIZATION_TYPES)
+            .map(type -> {
+                try {
+                    // noinspection unchecked
+                    Class<? extends ThreadInitializationFactory> clazz =
+                            (Class<? extends ThreadInitializationFactory>) Class.forName(type);
+                    return clazz.getDeclaredConstructor().newInstance();
+                } catch (ClassNotFoundException | NoSuchMethodException | InvocationTargetException
+                        | InstantiationException | IllegalAccessException e) {
+                    throw new IllegalArgumentException(
+                            "Error instantiating initializer " + type + ", please check configuration");
+                }
+            })
+            .collect(Collectors.toUnmodifiableList());
+
+    /**
+     * Chains configured initializers to run before/around any given runnable, returning a runnable intended to be run
+     * by a new thread.
+     */
+    static Runnable wrapRunnable(Runnable runnable) {
+        Runnable acc = runnable;
+        for (ThreadInitializationFactory INITIALIZER : INITIALIZERS) {
+            acc = INITIALIZER.createInitializer(acc);
+        }
+        return acc;
+    }
```

Comment:
Just to be sure I understand, the reason for this pattern is to allow for initializer-specific cleanup on exit? Otherwise, a pattern of iterative runnable invocations would seem less error-prone. I suppose either version allows arbitrary hijacking of the intended run method; the current pattern could skip calling the wrapped runnable, but the iterative approach could just never return or throw an exception on termination.

Reply:
It isn't that we need to do cleanup on the way out per se, but that we want to insert a stack frame in python. If we end up dropping this frame (which has some weird side effects when you "step out" of the apparent top-level frame), then we could just run one after the other, no wrapping.

The current impl has its option of how it wants to do it - I played with an API that would offer a DSL for either wrapping or prefixing, but didn't really love it, I can try bringing that back if you'd prefer.

Reply:
Okay, in light of the arm64 issues, a wrapper is necessary. The question then will be how we name that, if JavaThread is clear and simple enough, or if we want to use it to get some more info to the user.